data/dataset.py

```diff
@@ -0,0 +1,126 @@
+import os.path
+import itertools
+
+import torch
+import fortepyan as ff
+from tqdm import tqdm
+from tokenizer import Tokenizer
+from datasets import load_dataset
+from prepare_dataset import process_record
+
+from data.quantizer import MidiQuantizer
+
+
+class TokenizedMidiDataset:
+    def __init__(
+        self,
+        split: str = "train",
+        n_dstart_bins: int = 3,
+        n_duration_bins: int = 10,
+        n_velocity_bins: int = 10,
+        sequence_len: int = 128,
+        device: str = "cpu",
+    ):
+        self.sequence_len = sequence_len
+        self.device = device
+
+        self.quantizer = MidiQuantizer(
+            n_dstart_bins=n_dstart_bins,
+            n_duration_bins=n_duration_bins,
+            n_velocity_bins=n_velocity_bins,
+        )
+
+        self.tokenizer_src = Tokenizer(keys=["pitch", "dstart_bin"])
+        self.tokenizer_tgt = Tokenizer(keys=["duration_bin", "velocity_bin"])
+
+        self.dataset = load_dataset(path="roszcz/maestro-v1", split=split)
+
+        self.src_vocab, self.tgt_vocab = self._build_vocab()
+
+        self.processed_records = []
+        self.samples = []
+
+        self._build()
+
+    def _build(self):
+        self.processed_records = self.load_dataset()
+
```

Comment:
```suggestion
        self.processed_records = self.load_dataset()
        self.samples = self._build()

    def _build(self) -> list[Sample]:
        ...
```

I think it's better to be explicit about property creation in the `__init__`, than to hide it in methods